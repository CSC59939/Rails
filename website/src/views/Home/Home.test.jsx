import React from 'react';
import renderer from 'react-test-renderer';
import { Home } from '..';
import {
  shallow,
  configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {
  Button,
  Icon,
} from 'antd';

configure({ adapter: new Adapter() });

describe('Home Page', () => {
  it('Should render home page without any errors', () => {
    expect(renderer.create(<Home />).toJSON()).toMatchSnapshot();
  });
  it('should contains a h1 tag with class name Title with text Rails', () => {
    expect(shallow(<Home />).contains(<h1 className="title">Rails</h1>)).toBe(true);
  });
  it('should contains a button with classname home-button and link to signup', () => {
    const component = (
      <Button className="home-button" href="/signup">
        Get Started
        <Icon type="right-circle" theme="filled" />
      </Button>);
    expect(shallow(<Home />).contains(component)).toBe(true);
  });
  it('should have a class name home-button', () => {
    expect(shallow(<Home />).exists('.home-button')).toBe(true);
  });
});
