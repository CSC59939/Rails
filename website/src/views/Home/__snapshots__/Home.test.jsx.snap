// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Should render home page without any errors 1`] = `
<div
  className="home"
>
  <h1
    className="title"
  >
    Rails
  </h1>
  <h5
    className="tag-line"
  >
    Keep students on track
  </h5>
  <a
    className="ant-btn home-button"
    href="/signup"
    onClick={[Function]}
  >
    <span>
      Get Started
    </span>
    <i
      className="anticon anticon-right-circle"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="right-circle"
        fill="currentColor"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm154.7 454.5l-246 178c-5.3 3.8-12.7 0-12.7-6.5v-46.9c0-10.2 4.9-19.9 13.2-25.9L566.6 512 421.2 406.8c-8.3-6-13.2-15.6-13.2-25.9V334c0-6.5 7.4-10.3 12.7-6.5l246 178c4.4 3.2 4.4 9.8 0 13z"
        />
      </svg>
    </i>
  </a>
  <div
    className="ant-card about-card ant-card-bordered"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          About Rails
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <p>
        “Rails” is a platform for teachers and students. Rails will help organize all events in a class. From helping set due dates and reminders to providing a discussion forum for events in a class.
      </p>
    </div>
  </div>
</div>
`;
